portal-impl/src/com/liferay/portal/service/permission/UserPermissionImpl.java

```diff
@@ -167,8 +167,8 @@ public boolean contains(
 							return true;
 						}
 
-						// Organization administrators can only manage normal
-						// users
+						// Organization administrators and those with "Manage
+						// Users" permission can only manage normal users
 
 						if (!UserGroupRoleLocalServiceUtil.hasUserGroupRole(
```

Comment:
This can be moved out of the while loop since it's doing the same check for each iteration.